indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -95,66 +110,76 @@
 
             // Visible search field
             self.$searchInput = $('<input>', {
-                class: 'js-search-category js-typeahead',
+                class: 'js-search-category',
                 type: 'search',
                 name: 'q',
                 placeholder: $T.gettext("Search")
             }).attr('autocomplete', 'off');
-
-            // Insert elements in DOM
-            var $typeaheadForm = $($.parseHTML(self._typeaheadTemplate)[0]);
-            $typeaheadForm.find('.typeahead__query').append(self.$searchInput);
-            self.element.prepend($typeaheadForm);
-
-            // Typeahead init
-            self.$searchInput.typeahead({
-                dynamic: true,
-                filter: false,
-                source: {
-                    ajax: {
-                        url: build_url(Indico.Urls.Categories.search),
-                        path: 'categories',
-                        data: {
-                            q: "{{query}}"
-                        }
-                    }
-                },
-                resultContainer: false,
-                cancelButton: false,
-                display: 'title',
-                callback: {
-                    onClickAfter: function(node, a, item) {
-                        self.goToCategory(item.id);
-                    },
-                    onResult: function(node, query, result) {
-                        self._renderResults(result);
-                    },
-                    onHideLayout: function() {
-                        self.$categoryResults.html('');
-                        self.$category.show();
-                        self.$categoryTree.show();
-                    },
-                    onShowLayout: function(node, query) {
-                        self.$category.hide();
-                        self.$categoryTree.hide();
-                    }
+            self.element.prepend(self.$searchInput);
+
+            self.$searchInput.on('input', function() {
+                var $this = $(this);
+
+                if ($this.val().length > 1) {
+                    clearTimeout(self._delaySearchResults);
+                    self._delaySearchResults = setTimeout(function() {
+                        self._currentRequest = $.ajax({
+                            url: build_url(Indico.Urls.Categories.search),
+                            data: {q: $this.val()},
+                            beforeSend: function() {
+                                if(self._currentRequest != null) {
+                                    self._currentRequest.abort();
```

Comment:
It means you cannot apply regular error handling so actual errors wouldn't show the error dialog. Instead of aborting I'd rather keep track of what's the current request and ignore responses to others (e.g. by sending the search term back in the response and comparing it to the current one)